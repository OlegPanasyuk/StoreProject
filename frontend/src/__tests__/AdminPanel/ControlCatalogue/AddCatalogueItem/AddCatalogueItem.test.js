import React from 'react';
import AddCatalogueItemConnected, { AddCatalogueItem } from '../../../../AdminPanel/ControlCatalogue/AddCatalogueItem';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    getCatalogue: () => { },
    cancel: () => { },
    addCatalogueItem: () => { },
    addErrorToState: () => { },
    addItem: {
        show: false,
        name: '',
        description: '',
        parent_id: ''
    }
}) {
    let component = shallow(<AddCatalogueItem {...props} />);
    return { component, props };
}

describe('AddCatalogueItem --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(AddCatalogueItem.propTypes.addItem).toBe(PropTypes.object);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});


describe('AddCatalogueItem ', () => {
    let initialState = {
        adminPanel_catalogue: {
            addItem: {
                show: false,
                name: '',
                description: '',
                parent_id: ''
            }
        }
    };

    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><AddCatalogueItemConnected /></Provider>);
    });
    it('+++ render the connected(SMART) component',() => {
        expect(wrapper.find(AddCatalogueItemConnected).length).toEqual(1);
    });
});

