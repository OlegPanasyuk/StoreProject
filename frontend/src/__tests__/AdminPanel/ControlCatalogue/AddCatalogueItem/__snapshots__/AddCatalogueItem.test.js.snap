// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCatalogueItem --- Snapshot Component must be rendered 1`] = `
<Fragment>
  <Form
    as="form"
    inline={false}
  >
    <h4>
      Add item to catalogue: 
    </h4>
    <FormGroup
      as="div"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Name:
      </FormLabel>
      <FormControl
        as="input"
        defaultValue=""
        type="text"
      />
    </FormGroup>
    <FormGroup
      as="div"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        description:
      </FormLabel>
      <FormControl
        as="input"
        defaultValue=""
        type="text"
      />
    </FormGroup>
    <FormGroup
      as="div"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Parent:
      </FormLabel>
      <FormControl
        as="input"
        defaultValue=""
        type="number"
      />
    </FormGroup>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Save
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      Cancel
    </Button>
  </Form>
</Fragment>
`;
